Feature/HM-130-Admins-Add-heighborhoods-rating

petarbelchev opened the pull request:
Neighborhoods Rating functionality has been implemented.

petarbelchev committed:
Heighborhoods Rating functionality has been implemented.

rosteslav commented on src/Admins/Admins.Application/Features/Admins/Commands/AddNeighbourhoodsRating/AddNeighbourhoodsRatingCommandHandler.cs:
```diff
@@ -0,0 +1,13 @@
+        private readonly IAdminRepository _adminRepository = adminRepository;
+
+        public async Task Handle(AddNeighbourhoodsRatingCommand request, CancellationToken cancellationToken)
+            => await _adminRepository.AddNeighbourhoodsRating(request, cancellationToken);
```
repositories should not work based on commands, use `NeighbourhoodsRatingModel` we created in this review

petarbelchev replied:
Got it

petarbelchev committed:
Add GET /neighbourhoods/regions endpoint

rosteslav commented on src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs:
```diff
@@ -61,6 +65,47 @@ public async Task AddMultiplePropertiesFromCsvFile(IFormFile csvFile)
+                var secondary = new NeighbourhoodsRating
+                {
+                    Id = 2,
+                    ForLiving = JsonConvert.SerializeObject(command.ForLiving.ElementAt(1)),
```
.Last(). ElementAt(1) is not a good way

rosteslav commented on src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs:
```diff
@@ -61,6 +65,47 @@ public async Task AddMultiplePropertiesFromCsvFile(IFormFile csvFile)
+                    Luxury = JsonConvert.SerializeObject(command.Luxury.ElementAt(1))
+                };
+
+                if (_context.NeighbourhoodsRating.Any(n => n.Id == best.Id))
```
better delete
`await _context.NeighbourhoodsRating.ExecuteDeleteAsync(cancellationToken);`
all and just insert

rosteslav commented on src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs:
```diff
@@ -78,6 +123,51 @@ public async Task<IEnumerable<IdentityUser>> GetAllBrokers()
+
+            try
+            {
+                var ratings = await _context.NeighbourhoodsRating.Where(n => n.Id == 1 || n.Id == 2).ToArrayAsync(cancellationToken);
```
just all, without where

rosteslav commented on src/Admins/Admins.Infrastructure/Repositories/AdminRepository.cs:
```diff
@@ -78,6 +123,51 @@ public async Task<IEnumerable<IdentityUser>> GetAllBrokers()
+                var ratings = await _context.NeighbourhoodsRating.Where(n => n.Id == 1 || n.Id == 2).ToArrayAsync(cancellationToken);
+                var result = new NeighbourhoodsRatingModel
+                {
+                    ForLiving = ratings.Select(r => JsonConvert.DeserializeObject<string[]>(r.ForLiving)),
```
isn't that `IEnumerable<IEnumerable<string>>`, why are you Deserializing it to <string[]>?
did you test it? Does it work?

petarbelchev replied:
It is working, however there is no reason to be string[]

petarbelchev committed:
Improve the logic for updating neighborhood ratings;
Use model instead of command in Repository;
Deserialize object into IEnumerable instead of array.
